src/main/java/org/folio/inventory/resources/Items.java

```diff
@@ -107,6 +115,57 @@ public void register(Router router) {
           routingContext, webContext));
   }
 
+  private void move(RoutingContext routingContext) {
+
+    WebContext context = new WebContext(routingContext);
+    JsonObject itemsMoveJsonRequest = routingContext.getBodyAsJson();
+
+    Optional<ValidationError> validationError = itemsMoveHasRequiredFields(itemsMoveJsonRequest);
+    if (validationError.isPresent()) {
+      unprocessableEntity(routingContext.response(), validationError.get());
+    } else {
+      ItemsMove itemsMove = MoveUtil.jsonToItemsMove(itemsMoveJsonRequest);
+      List<String> itemIdsToUpdate = itemsMove.getItemIds();
+      try {
+        storage.getItemCollection(context).findByCql(
+          buildQueryByIds(itemIdsToUpdate),
+          PagingParameters.defaults(),
+          success -> updateManyItemsWithNewHoldingsRecordId(routingContext, context, success.getResult(), itemsMove.getToHoldingsRecordId(), itemIdsToUpdate),
+          FailureResponseConsumer.serverError(routingContext.response()));
+      } catch (UnsupportedEncodingException e) {
+        ServerErrorResponse.internalError(routingContext.response(), e.toString());
+      }
+    }
+  }
+
+  private void updateManyItemsWithNewHoldingsRecordId(
+    RoutingContext routingContext,
+    WebContext context,
+    MultipleRecords<Item> itemsToUpdate,
+    String toHoldingsRecordId,
+    List<String> itemIdsToUpdate) {
+
+    ItemCollection storageItemCollection = storage.getItemCollection(context);
+
+    List<CompletableFuture<Item>> updates = itemsToUpdate.records.stream()
```

Comment:
Do we need some additional validation, e.g.?:
* Is holdings record exists;
* Is it allowed to move to a holdings from an another instance;
* I am not sure if it is ok to move a MARC controlled item, maybe @OleksiiKuzminov can clarify.

Reply:
We almost certainly should stop an attempt to move to a holdings record that does not exist

Reply:
>  am not sure if it is ok to move a MARC controlled item

That is an excellent requirements question

Reply:
> I am not sure if it is ok to move a MARC controlled item

What was the answer to this question?

Reply:
Asked Ann-Marie in jira.

Reply:
Response:
> Hi Bohdan Suprun SRS does not track which holdings or instance an item is linked to, so there should not be any issues with moving an item from one holdings to another, or from one instance to another.
>
> It will be slightly different when you get to the story of moving a holdings from one instance to another, but that's not in this story.

Reply:
So, we can merge it as is. Am I correct?

Reply:
Yes. We should keep in mind this restriction for holdings move feature.